### PR TITLE
Implemented hyperfocal distance to utils.py

### PR DESCRIPTION
**Pull request by jremes-foss**
Hyperfocal distance (ie. distance between a camera lens and the closest object which is in focus when the lens is focused at infinity) implemented to sympy/physics/optics/utils.py. Added unit tests to respective test file.

### CODE REVIEW

File: sympy/physics/optics/utils.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 * lens_makers_formula
 * mirror_formula
 * lens_formula
+* hyperfocal_length
```
Comment (thisch) [inline sympy/physics/optics/utils.py]: this should be hyperfocal_distance

File: sympy/physics/optics/utils.py [hunk 2]
```diff
@@ -15,6 +16,7 @@
            'lens_makers_formula',
            'mirror_formula',
            'lens_formula'
+           'hyperfocal_length'
```
Comment (thisch) [inline sympy/physics/optics/utils.py]: this as well

**Comment by Newman101** [comment]
Ah, thanks for notifying. I'll fix this.

**Comment by Newman101** [comment]
I think I just left the changes unsaved, sorry for confusion.

**Comment by Newman101** [comment]
That's fixed now.

**Comment by Newman101** [comment]
The build failed for some reason. Just in case, I rounded the unit test to two decimals. If that won't work I will see where it goes wrong tomorrow morning.

**Comment by Newman101** [comment]
All checks passed.

File: sympy/physics/optics/utils.py [hunk 3]
```diff
@@ -422,3 +424,31 @@ def lens_formula(focal_length=None, u=None, v=None):
         if focal_length == oo:
             return Limit(u*_f/(u + _f), _f, oo).doit()
         return u*focal_length/(u + focal_length)
+
+def hyperfocal_distance(f = None, N = None, c = None):
```
Comment (asmeurer) [inline sympy/physics/optics/utils.py]: Why do you have defaults set for these? The function won't work correctly unless all three are given.

Comment (Newman101) [inline sympy/physics/optics/utils.py]: I've removed the defaults. Running the tests locally to see it works fine before committing.

**Pull request merged**